### DISCUSSION TITLE
MCCTEAM AUTORECONNECT

### DISCUSSION THREAD

**Discussion by NivellI** [ORelio/Minecraft-Console-Client#2492] [General]
Hello, I have been using mcc for a long time, but it kicks me out of the game at certain times on the server I entered. And I'm struggling with manually re-inserting accounts. How can I make this automatic?
(I am currently using the latest version of mcc team.)
afk i'm tired of putting it back manually
please come up with a solution
thank you.

**Comment by breadbyte** [comment]
Enable the AutoRelog in the config file using these settings.

![image](https://user-images.githubusercontent.com/14045257/235413801-631827d3-689a-4503-bbf7-4ed12f61d83f.png)

**Reply by NivellI** [reply]
It's not exactly what I'm looking for.
How do I automatically print a message to the server after autorelog?

**Reply by NivellI** [reply]
I used this, it reconnected without any problems, but /login does not automatically throw it, how can you suggest a solution?

**Reply by breadbyte** [reply]
> but /login does not automatically throw it, how can you suggest a solution?

Use the ScriptScheduler to send a `/login` command after logging in.
![image](https://user-images.githubusercontent.com/14045257/235441119-b8418659-472d-4b69-95bb-b62e757b335b.png)

**Reply by NivellI** [reply]
Can you send a sample draft photo, I activated it but there is no result

**Reply by NivellI** [reply]
And I don't need a timer. I want them to log back to the server right after they've removed the afk from the server, what do I need to write for this?

**Reply by breadbyte** [reply]
> And I don't need a timer. I want them to log back to the server right after they've removed the afk from the server, what do I need to write for this?

The screenshots provided don't use timers and have them disabled (`Enable = false`).
It executes the `send /login pass` action whenever you have connected to the server.

**Reply by milutinke** [reply]
> And I don't need a timer. I want them to log back to the server right after they've removed the afk from the server, what do I need to write for this?

You can't do it without a delay, just put 1 second.
As for sending the login command, just use the task from above but instead of doing `Action = "send /login pass"`, set it to `Action = "script login.txt"`, make a new file can call it "login.txt", in it put the following:
```
wait 5
send /login yourpasswordhere
```
Sometimes a slight delay is needed in order to send commands and log in
Also, see if your server has the `/remember` command, so you can skip logging in.

if you are supposed to join a sub server, check if there is a `/server` command, and the name of the sub server, usually on multi server serves there is that command, so you could add the following at the end of the file after the `send /login` command:
```
wait 2
send /server subservernamehere
```

**Reply by NivellI** [reply]
How can I switch to a sub server, namely skyblock, in login.txt after /login? Can you tell me what should I write?

do i have to write it down?
for example ;

wait 3
send /login pass
wait 5
send /skyblock

If I write this way, will the code work?

**Reply by breadbyte** [reply]
Yes, that should work.

**Reply by NivellI** [reply]
I have another question for you
Can I place an afk account on a server via discord? is there a detailed explanation video or a bibliography?

**Reply by milutinke** [reply]
> I have another question for you
> Can I place an afk account on a server via discord? is there a detailed explanation video or a bibliography?
> 

You only can send the commands via discord once the client is on the server, not before.
Since chat bots load after you connect to the server.

**Reply by NivellI** [reply]
I didn't understand anything they said. Can I put afk on my server using Discord? is there such a way?

**Reply by breadbyte** [reply]
You cannot login to the server using discord.
You can control the client from discord once you have logged in.

**Comment by 99marvel** [comment]
Hi, I have an adjacent question. (maybe it should be it's own post but I'd like to reduce clutter) 

I'm using the auto reconnect feature exactly as described above, to reconnect after a nightly reboot. I have the string "Server Shutting Down" as the only trigger phrase for reconnecting, which is the message sent on reboot. 

This works great... about 7/8 times. As far as I know the code running on the server (and therefore the message sent to client) is exactly the same every time, but occasionally MCC doesn't log back in. Is there a known fix for this or a suggested workaround? Thanks! :)

**Reply by milutinke** [reply]
Try with increasing the time it takes to re-connect.
